test_runner/fixtures/pageserver/allowed_errors.py

```diff
@@ -0,0 +1,118 @@
+        ".*gc_loop.*Gc failed, retrying in.*: Cannot run GC iteration on inactive tenant",  # Tenant::gc precondition
+        ".*compaction_loop.*Compaction failed.*, retrying in.*timeline or pageserver is shutting down",  # When compaction checks timeline state after acquiring layer_removal_cs
+        ".*query handler for 'pagestream.*failed: Timeline .* was not found",  # postgres reconnects while timeline_delete doesn't hold the tenant's timelines.lock()
+        ".*query handler for 'pagestream.*failed: Timeline .* is not active",  # timeline delete in progress
+        ".*task iteration took longer than the configured period.*",
+        # this is until #3501
+        ".*Compaction failed.*, retrying in [^:]+: Cannot run compaction iteration on inactive tenant",
+        # these can happen anytime we do compactions from background task and shutdown pageserver
+        r".*ERROR.*ancestor timeline \S+ is being stopped",
+        # this is expected given our collaborative shutdown approach for the UploadQueue
+        ".*Compaction failed.*, retrying in .*: Other\\(queue is in state Stopped.*",
+        # Pageserver timeline deletion should be polled until it gets 404, so ignore it globally
+        ".*Error processing HTTP request: NotFound: Timeline .* was not found",
+        ".*took more than expected to complete.*",
+        # these can happen during shutdown, but it should not be a reason to fail a test
+        ".*completed, took longer than expected.*",
+        # AWS S3 may emit 500 errors for keys in a DeleteObjects response: we retry these
+        # and it is not a failure of our code when it happens.
+        ".*DeleteObjects.*We encountered an internal error. Please try again.*",
+    ]
+
+
+if __name__ == "__main__":
+    import argparse
+
+    def check_allowed_errors_impl(_args):
+        import sys
+
+        allowed_errors = default_pageserver_allowed_errors()
+
+        # add any test specifics here; cli parsing is not provided for the
+        # difficulty of copypasting regexes as arguments without any quoting
+        # errors.
+
+        errors = scan_pageserver_log_for_errors(sys.stdin, allowed_errors)
+
+        for lineno, error in errors:
+            print(f"-:{lineno}: {error.strip()}", file=sys.stderr)
+
+        print(f"\n{len(errors)} not allowed errors", file=sys.stderr)
+
+        if len(errors) > 0:
+            sys.exit(1)
+        sys.exit(0)
+
+    parser = argparse.ArgumentParser(
+        "neon_fixtures", description="do some checks done in test suite on demand"
+    )
+    subparsers = parser.add_subparsers()
+
+    check_allowed_errors = subparsers.add_parser(
+        "check_allowed_errors", help="check stdin against pageserver global allowed_errors"
+    )
+    check_allowed_errors.set_defaults(func=check_allowed_errors_impl)
+
+    args = parser.parse_args()
+    args.func(args)
```

Comment:
I think the usage of subparsers with `check_allowed_errors_impl ` looks a bit complicated here, can we inline the function instead (instead of using subparsers)?

Reply:
5b93350 dropped the subparsers, still a separate func for dispatch and argparse for help.

Does it look good enough?